python/ray/autoscaler/gcp/tags.py

```diff
@@ -0,0 +1,35 @@
+"""The Ray autoscaler uses tags/labels to associate metadata with instances.
+
+Notice that the name 'tag' here is somewhat misnomer, since gcp uses the name
+'label' for key-value pairs, and 'tag' for value-only tags.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+TAG_KEYS = {
```

Comment:
One possibility here is to use the GCP labeling conventions for AWS (I assume they will work for AWS, or we could find some common supported subset). That would be an API breaking change but I think that's fine at this stage of the project if it means cleaner code.

Reply:
That could work too. Do you think I should make the change right now? I'm leaning towards leaving it like this and rethinking the tagging system later when more node providers exist.

Reply:
How about switching to characters and underscores only? That should be sufficiently conservative for any future systems.

I do think it's worth cleaning it up to have one tag set instead of multiple.

Reply:
Haha, I think gcp doesn't actually support underscores, neither capital letters. I think the only allowed characters are `a-z`, `0-9`, and `-`.

I'm happy to change all the tags to follow this spec if that sounds good to you all.

Reply:
I was wrong. gcp tags do support underscores. I removed the double-tagging system and changed all the tags to work with gcp spec (`[a-z0-9_-]`).